core/src/main/scala/org/apache/spark/shuffle/FileShuffleBlockResolver.scala

```diff
@@ -23,6 +23,8 @@ import java.util.concurrent.atomic.AtomicInteger
 
 import scala.collection.JavaConversions._
 
+import com.google.common.annotations.VisibleForTesting
```

Comment:
Hmmm... I wasn't familiar with this.  It's an interesting idea, but I'm left wondering why we would want to use specifically Google's annotation, and why we would want to use it in this particular file.  I'm open to the notion of this kind of "for testing" marker, and even to adding it in lots of other places in the Spark code.  I'm just not clear about the particular choices you've made here.